### PR TITLE
TTS enabled behind 'CSDTTS' flag for CSD/CSP

### PR DESCRIPTION
**Pull request by epeach**
Added TTS in CSD and CSP. Accessible behind 'CSDTTS' flag. Updated styling for Inline Audio button.

Old InlineAudio (CSF):
image

New InlineAudio (CSF):
csfnewaudio

New InlineAudio (CSD/CSP):
csdnewaudio

### CODE REVIEW

**Comment by joshlory** [comment]
+cc @Hamms — I think there might be an extra generation step to get the TTS files uploaded to S3.

File: apps/src/templates/instructions/TopInstructionsCSP.jsx [hunk 1]
```diff
@@ -173,11 +181,17 @@ var TopInstructions = React.createClass({
...
     ];
+    const ttsUrl = this.props.ttsMarkdownInstructionsUrl;
 
     return (
       <div style={mainStyle} className="editor-column">
         <PaneHeader hasFocus={false}>
+
           <div style={styles.paneHeaderOverride}>
+            {experiments.isEnabled('CSDTTS') &&
+            <div id="foundIt!" style={styles.audioControls}>
```
Comment (Hamms) [inline apps/src/templates/instructions/TopInstructionsCSP.jsx]: nit: exclamation marks in ids, although technically allowed by HTML5 (but not 4), are still a little weird.

Comment (epeach) [reply apps/src/templates/instructions/TopInstructionsCSP.jsx]: Whoops! That's a hold-over from debugging things. It's already out in the newest version.

**Comment by epeach** [comment]
@Hamms PTAL. Styling changed across all InlineAudio - okay-ed by Poorva and Ryan.

File: dashboard/app/helpers/levels_helper.rb [hunk 2]
```diff
@@ -294,6 +294,16 @@ def weblab_options
...
 
+    script = @script
+
+    # Text to speech
+    if script && script.text_to_speech_enabled?
+      level_options['ttsInstructionsUrl'] = @level.tts_url(@level.tts_instructions_text)
+      level_options['ttsMarkdownInstructionsUrl'] = @level.tts_url(@level.tts_markdown_instructions_text)
+    end
+
+    app_options[:textToSpeechEnabled] = @script.try(:text_to_speech_enabled?)
```
Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb]: I'd love to see this logic pulled out of both here and blockly_options into a shared method

File: apps/src/templates/instructions/TopInstructionsCSP.jsx [hunk 3]
```diff
@@ -238,7 +262,9 @@ module.exports = connect(function propsFromStore(state) {
     markdown: state.instructions.longInstructions,
     noVisualization: state.pageConstants.noVisualization,
     collapsed: state.instructions.collapsed,
-    documentationUrl: state.pageConstants.documentationUrl
+    documentationUrl: state.pageConstants.documentationUrl,
+    ttsInstructionsUrl: state.pageConstants.ttsInstructionsUrl,
```
Comment (Hamms) [inline apps/src/templates/instructions/TopInstructionsCSP.jsx]: nit: it doesn't look like we're actually using the Instructions url, just the MarkdownInstructions

File: apps/src/templates/instructions/InlineAudio.jsx [hunk 4]
```diff
@@ -169,22 +172,35 @@ const InlineAudio = React.createClass({
...
+              style={[styles.buttonImg,
+                (this.props.style && this.props.style.buttonImg) ? this.props.style.buttonImg : null]}
+            />
           </div>
-          <button style={[styles.button, styles.playPauseButton]} onClick={this.toggleAudio}>
-            <img
-              style={styles.buttonImg}
-              src={this.state.playing ?
-                this.props.assetUrl("media/common_images/pause.png") :
-                this.props.assetUrl("media/common_images/play.png")}
```
Comment (Hamms) [inline apps/src/templates/instructions/InlineAudio.jsx:181]: we can remove these images!

File: apps/src/templates/instructions/InlineAudio.jsx [hunk 5]
```diff
@@ -169,22 +172,35 @@ const InlineAudio = React.createClass({
   render: function () {
     if (this.props.textToSpeechEnabled && !this.state.error && this.isLocaleSupported() && this.getAudioSrc()) {
       return (
-        <div className="inline-audio">
-          <div style={[styles.button, styles.volumeButton]}>
-            <img style={styles.buttonImg} src={this.props.assetUrl("media/common_images/volume.png")} />
+        <div
+          className="inline-audio"
+          style={(this.props.style && this.props.style.wrapper) ? this.props.style.wrapper : null}
```
Comment (Hamms) [inline apps/src/templates/instructions/InlineAudio.jsx]: I believe you can just do `style={this.props.style && this.props.style.wrapper}`, here and elsewhere

**Comment by epeach** [comment]
@Hamms Made the requested changes and included the images in the PR description. PTAL. Thanks!

**Pull request merged**